cardano-api/src/Cardano/Api/Governance/Poll.hs

```diff
@@ -427,15 +312,10 @@ verifyPollAnswer poll answer witness = do
       , invalidAnswerAcceptableAnswers
       }
 
-  unless isValid $
-    Left ErrGovernancePollInvalidWitness
+  pure [ witVKeyHash wit | (ShelleyKeyWitness _ wit) <- getTxWitnesses tx ]
```

Comment:
That's a good point. We could potentially omit normal unrelated payment keys this way. Ideally, I would print those hashes as bech32 `poolId` but we can't actually have any guarantee that these are indeed `poolId`. Yet, it would also make it clear that one has to check against the existing distribution of pools to assess their validity and non-existing pool ids would be discarded.

Reply:
Should we report all witness but group the pool ones separately?

```
{ "pollWitnesses":
  { "pools": [ ... ]
  , "other": [..]
  }
}
```

Reply:
> @Jimbo4350 You could use executeQueryCardanoMode with the QueryStakePools query to return the pool ids 

I'll push back on that one for three reasons:

1. It is slow.
    Few seconds to minutes, so this won't scale well when we need to verify thousands of answers in a row. 
    
2. It makes those commands no longer 'static'
    This introduces a dependency on a running cardano-node, which makes it particularly painful as go we go from a relatively easy-to-use static command even usable in an air-gapped environment, to something a lot heavier with a full node. 
    
3. Verification is done against a fixed snapshot  
    We need to verify pool ids against a known snapshot performed ahead of time. So even if some pool ids is registered _now_, doesn't make it valid at the moment the verification is performed. Snapshots are produced externally and thus, we can reliably do that verification in this step (unless we provide the full snapshot to the command but I am not looking into that. It's way easier for every ends as currently implemented).
    
> @newhoggy Should we report all witness but group the pool ones separately?

Not a huge fan. It requires an extra data structure which feels oddly arbitrary for that purpose. Plus, we really have no use of the "other" witnesses and we can't be sure that "pools" are indeed pools. So I think it's best to only go with @dcoutts' suggestion and (a) return only extra signatories, (b) fails when there's no extra signatories as we always necessarily expect at least one.